SDWebImage/Swift/SDWebImage.swift

```diff
@@ -0,0 +1,40 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ * (c) Chester Liu <https://github.com/skyline75489>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import Foundation
+
+#if os(macOS)
+    import AppKit
+    public typealias SDImageType = NSImage
+    public typealias SDColorType = NSColor
+    public typealias SDImageViewType = NSImageView
+    typealias Button = NSButton
+#else
+    import UIKit
+    public typealias SDImageType = UIImage
+    public typealias SDColorType = UIColor
+    #if !os(watchOS)
+        public typealias SDImageViewType = UIImageView
+    #endif
+#endif
+
+extension SDImageViewType {
+
+    @nonobjc public func sd_setImage(url: URL?, placeholderImage: SDImageType? = nil, options: SDWebImageOptions = .init(rawValue: 0), progress: SDWebImageDownloaderProgressBlock? = nil, completed: SDExternalCompletionBlock? = nil) {
```

Comment:
Here `@nonobjc` prevents this Swift file from being imported to `SDWebImage-swift.h` (for Swift -> Obj-C), causing compilation error. This should be OK because you don't actually need this if you're using Obj-C.